tests/test_meta_functions_against_registry.py

```diff
@@ -163,6 +187,43 @@ def test_get_underlying_coins(metaregistry, registries, max_pools, curve_api):
             assert actual_output == metaregistry_output
 
 
+def test_get_underlying_balances(metaregistry, registries, max_pools, curve_api):
+    for i, registry in enumerate(registries):
+        total_pools = registry.pool_count() if max_pools == 0 else max_pools
+        for pool_index in range(total_pools):
+            metaregistry_output = None
+            pool = registry.pool_list(pool_index)
+            metaregistry_borks = False
+            metaregistry_output = metaregistry.get_underlying_balances(pool)
+
+            if i in [
+                METAREGISTRY_STABLE_FACTORY_HANDLER_INDEX,
+                METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX,
+            ]:
+                registry_query_borks = False
```

Comment:
Is there a reason to use try/catch instead of brownie.reverts() here?

Reply:
Because we're trying to get the correct values to check results against, and the underlying registries have several issues which the metaregistry doesn't. So in order to check if the metaregistry outputs make sense, its kind of needed to go through registry, try something and if it reverts try something else. Sometimes metaregistry can give a more correct output than the underlying registry as well. So these edge cases need to be considered to get actual outputs.